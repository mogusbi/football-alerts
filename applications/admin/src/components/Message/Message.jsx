import CloseIcon from '@material-ui/icons/Close';
import IconButton from '@material-ui/core/IconButton';
import Snackbar from '@material-ui/core/Snackbar';
import PropTypes from 'prop-types';
import React, {memo} from 'react';

function Message ({closeHandler, message, open}) {
  return (
    <Snackbar
      action={(
        <IconButton
          aria-label='Close'
          color='inherit'
          onClick={closeHandler}
          key='close'
        >
          <CloseIcon />
        </IconButton>
      )}
      anchorOrigin={{
        horizontal: 'left',
        vertical: 'bottom'
      }}
      autoHideDuration={6000}
      ContentProps={{
        'aria-describedby': 'message-id'
      }}
      message={(
        <span id='message-id'>
          {message}
        </span>
      )}
      onClose={closeHandler}
      open={open}
    />
  );
}

Message.propTypes = {
  closeHandler: PropTypes.func.isRequired,
  message: PropTypes.string.isRequired,
  open: PropTypes.bool.isRequired
};

export default memo(Message);
