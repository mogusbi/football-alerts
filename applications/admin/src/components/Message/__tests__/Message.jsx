import {shallow} from 'enzyme';
import React from 'react';
import Message from '../Message';

describe('Message', () => {
  let closeHandler;
  let message;
  let wrapper;

  beforeEach(() => {
    closeHandler = jest.fn();
    message = 'This is a test message';
  });

  describe('when open', () => {
    beforeEach(() => {
      wrapper = shallow(
        <Message
          closeHandler={closeHandler}
          message={message}
          open
        />
      );
    });

    it('should render component', () => {
      expect(wrapper.dive()).toMatchSnapshot();
    });
  });

  describe('when closed', () => {
    beforeEach(() => {
      wrapper = shallow(
        <Message
          closeHandler={closeHandler}
          message={message}
          open={false}
        />
      );
    });

    it('should render component', () => {
      expect(wrapper.dive()).toMatchSnapshot();
    });
  });
});
