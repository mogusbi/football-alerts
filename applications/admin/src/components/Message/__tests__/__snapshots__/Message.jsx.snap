// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message when closed should render component 1`] = `
<Snackbar
  ContentProps={
    Object {
      "aria-describedby": "message-id",
    }
  }
  TransitionComponent={[Function]}
  action={
    <WithStyles(IconButton)
      aria-label="Close"
      color="inherit"
      onClick={[MockFunction]}
    >
      <pure(CloseIcon) />
    </WithStyles(IconButton)>
  }
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={6000}
  classes={
    Object {
      "anchorOriginBottomCenter": "MuiSnackbar-anchorOriginBottomCenter-3",
      "anchorOriginBottomLeft": "MuiSnackbar-anchorOriginBottomLeft-7",
      "anchorOriginBottomRight": "MuiSnackbar-anchorOriginBottomRight-5",
      "anchorOriginTopCenter": "MuiSnackbar-anchorOriginTopCenter-2",
      "anchorOriginTopLeft": "MuiSnackbar-anchorOriginTopLeft-6",
      "anchorOriginTopRight": "MuiSnackbar-anchorOriginTopRight-4",
      "root": "MuiSnackbar-root-1",
    }
  }
  disableWindowBlurListener={false}
  message={
    <span
      id="message-id"
    >
      This is a test message
    </span>
  }
  onClose={[MockFunction]}
  open={false}
  transitionDuration={
    Object {
      "enter": 225,
      "exit": 195,
    }
  }
/>
`;

exports[`Message when open should render component 1`] = `
<Snackbar
  ContentProps={
    Object {
      "aria-describedby": "message-id",
    }
  }
  TransitionComponent={[Function]}
  action={
    <WithStyles(IconButton)
      aria-label="Close"
      color="inherit"
      onClick={[MockFunction]}
    >
      <pure(CloseIcon) />
    </WithStyles(IconButton)>
  }
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={6000}
  classes={
    Object {
      "anchorOriginBottomCenter": "MuiSnackbar-anchorOriginBottomCenter-3",
      "anchorOriginBottomLeft": "MuiSnackbar-anchorOriginBottomLeft-7",
      "anchorOriginBottomRight": "MuiSnackbar-anchorOriginBottomRight-5",
      "anchorOriginTopCenter": "MuiSnackbar-anchorOriginTopCenter-2",
      "anchorOriginTopLeft": "MuiSnackbar-anchorOriginTopLeft-6",
      "anchorOriginTopRight": "MuiSnackbar-anchorOriginTopRight-4",
      "root": "MuiSnackbar-root-1",
    }
  }
  disableWindowBlurListener={false}
  message={
    <span
      id="message-id"
    >
      This is a test message
    </span>
  }
  onClose={[MockFunction]}
  open={true}
  transitionDuration={
    Object {
      "enter": 225,
      "exit": 195,
    }
  }
/>
`;
